sys/posix/sockets/posix_sockets.c

```diff
@@ -959,7 +959,13 @@ static ssize_t socket_sendto(socket_t *s, const void *buffer, size_t length,
 #endif
 #ifdef MODULE_SOCK_UDP
         case SOCK_DGRAM:
-            if ((res = sock_udp_send(&s->sock->udp, buffer, length, &ep)) < 0) {
+            if (address == NULL) {
+                res = sock_udp_get_remote((sock_udp_t *)s->sock, &ep);
```

Comment:
Using `s->sock->udp` saves you the cast.

Reply:
Also (but we can do this also in a follow-up), how about using a wrapper (e.g. `_sock_get_remote()`)? This way `SOCK_RAW` also would benefit.

Reply:
The cast is now fixed.

I did not see a point in supporting connected raw sockets so far. Where would you like the wrapper function to live?

Reply:
> The cast is now fixed.

It's hard to track your changes during the review if you force-push (especially if it involves a rebase). So please only force-push (after squashing your commits appropriately), when the review is finished and the PR is approved.

> I did not see a point in supporting connected raw sockets so far. Where would you like the wrapper function to live?

Then let's move this as suggested to a follow-up.

Reply:
My apologies for that, will do better.